### ISSUE TITLE
[stable/datadog] livenessProbe fails when using host_network:true

### ISSUE THREAD

**Issue by elliottgorrell** [helm/charts#10953]
**Is this a request for help?**: no

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**:
Client: &version.Version{SemVer:"v2.12.2", GitCommit:"7d2b0c73d734f6586ed222a567c5d103fed435be", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.2", GitCommit:"7d2b0c73d734f6586ed222a567c5d103fed435be", GitTreeState:"clean"}

**Which chart**:
stable/datadog

**What happened**:
Liveness probe for the agent is broken when using `host_network: true`. The pod goes into an infinite termination cycle.

Caused by the changes introduced in #10827 

**What you expected to happen**:
The liveness probe to get a 200 and keep the container running

**How to reproduce it** (as minimally and precisely as possible):
Launch the chart with `--set daemonset.useHostNetwork="true"`. 

**Anything else we need to know**:
This is most likely because the port binding needs to be performed on 5555 like has been done for the statsd and trace ports in `daemonset.yml` when using host_network:true

I will create a PR to address this when I get home from work today :)

**Comment by willyyang** [comment]
hey did you get around to that PR ? :)

**Comment by xvello** [comment]
Hi @elliotkennedy , @willyyang 

I cannot reproduce with version `1.18.0` of the chart by running:

`helm install --set daemonset.useHostNetwork="true" --set datadog.apiKey=$(cat ~/.apikey) --name hostnet  stable/datadog`

```
NAME                                         READY   STATUS    RESTARTS   AGE
hostnet-datadog-652hv                        1/1     Running   0          8m
hostnet-datadog-9rxmz                        1/1     Running   0          8m
hostnet-datadog-psjtw                        1/1     Running   0          8m
hostnet-kube-state-metrics-5fc769455-vw55s   1/1     Running   0          8m
```

The health port on 5555 was added in Agent version 6.8.0, which made the chart 1.17.0 fail on deploy, as it still pinned the agent image 6.6.0.
[Version 1.18.0 updates the agent to 6.9.0](https://github.com/helm/charts/pull/10959/files), could you please make sure this is the version you are using?

Cheers

**Comment by elliotkennedy** [comment]
@xvello - wrong Elliot/t paging @elliottgorrell

**Comment by elliottgorrell** [comment]
@xvello @willyyang Sorry got busy but can confirm this was fixed in #10876 .
The PR im referencing was `1.17.0` of the chart so this was fixed by someone alot quicker than me :P in 1.18.0.

Thanks everyone!
